### DISCUSSION TITLE
PF Coupling with Electrostatic Energy

### DISCUSSION THREAD

**Discussion by jsevic** [idaholab/moose#25489] [Q&A Modules: Phase field]
Hello,

I am doing PF of simulation of spinodal decomposition interacting with an electrostatic energy term, similar to the derivation of the weak form for SplitCHParsed.

How do I couple an electrostatic energy term to the CH equation? Is it an additional kernel or is there a clever way to add it to the SplitCHParsed kernel for free-energy density? It looks like I could specify the derivative and then add directly to the free-energy density?

Thank you.

John Sevic

**Comment by dschwen** [comment]
I'd solve for the electrostatic potential using a poisson equation, and then couple that in the free energy expression

**Comment by hugary1995** [comment]
You can add the electrostatic term to the free energy. You free energy will then additionally depend on the electric potential, and this is taken care of by the derivative parsed materials. You electrostatics equation need to couple with the phase field, and that's the part you need to take care of.

**Reply by dschwen** [reply]
The charge density in the poisson equation will be a function of the PF concentrations.

**Reply by jsevic** [reply]
OK thanks let me show what I have so far and see if this makes sense.

My free-energy functional (immediately following eq. 2 on the P-F equation page) has the two usual terms, free-energy density and interface energy, and these are in eV. To this functional I add a term for coupling the electric potential energy, this term in eV is E_d = q*c*V, where q is charge and c is concentration variable and V is electric potential.

Now SplitCHParsed kernel calls DerivativeParsedMaterial, below is what I currently have without coupling to V.

Then in "expression" would I then add something like E_d = ev_J*c*V. Then DerivativeParsedMaterial automatically computes the partial derivative of E_d with respect to c as shown in the weak form table?

Thank you!

John Sevic


  [./local_energy]
    # Local free energy function in eV.
    type = DerivativeParsedMaterial
    property_name = f_loc
    coupled_variables = 'c T V'
    constant_names = 'a0  a2  a4  c1  c2  ev_J'
    constant_expressions = '30.0 -4.00 2.80
                            0.30 0.80
                            1.6e-19 '

    expression = 'ev_J*( (a0/2)*(0.0012*(T - 273) - c1)^2 -
      (a2/2)*(0.0012*(T - 273) - c2)*(4*c - 2)^2 -
      (a4/4)*(4*c - 2)^4 +
      0.25*(4*c - 2)^6 ) + INSERT POTENTIAL ENERGY TERM HERE'
    derivative_order = 2
    enable_jit = true
  [../]
